packages/EasyActivity/src/Bridge/Symfony/Serializers/CircularReferenceHandler.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EonX\EasyActivity\Bridge\Symfony\Serializers;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Exception;
+use ReflectionProperty;
+
+final class CircularReferenceHandler implements CircularReferenceHandlerInterface
+{
+    /**
+     * @var \Doctrine\ORM\EntityManagerInterface
+     */
+    private $entityManager;
+
+    public function __construct(EntityManagerInterface $entityManager)
+    {
+        $this->entityManager = $entityManager;
+    }
+
+    /**
+     * @noinspection BadExceptionsProcessingInspection
+     */
+    public function __invoke(object $object, string $format, array $context)
+    {
+        $className = \get_class($object);
+        try {
+            $identifier = $this->entityManager->getClassMetadata($className)
+                ->getSingleIdentifierFieldName();
+            $reflectionProperty = new ReflectionProperty($className, $identifier);
+            $reflectionProperty->setAccessible(true);
+
+            return \sprintf('%s#%s (circular reference)', $className, $reflectionProperty->getValue($object));
```

Comment:
`%s#%s (circular reference)` - this value will be stored in the database.
What about returning just `$reflectionProperty->getValue($object)` by default and `null` for exceptional case? It will be easier to work with such values.

Reply:
@mvlad as we just discussed with @akl-95, the idea of this new feature is to highlight a potential issue in the configs. It would be great to never store circular references in logs, so when this happens, a developer or an admin sees this in the logs and makes sure the configs are changed to exclude the corresponding relationship.